pkg/store/proxy.go

```diff
@@ -503,6 +531,10 @@ func startStreamSeriesSet(
 			}
 
 			if series := rr.r.GetSeries(); series != nil {
+				if applySharding && !shardMatcher.MatchesZLabels(series.Labels) {
```

Comment:
Is this placed at the correct line to not interfere with cancellations?

Reply:
I think so, this should make it so that not matching series are skipped